portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,26 +28,34 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private ArrayList<String> messages;
-
-  @Override
-  public void init() {
-    messages = new ArrayList<>();
-    messages.add("Bonjour, ca va?");
-    messages.add("Nda aga i di?");
-    messages.add("Hey, how are you?");
-  }
+  private ArrayList<String> comments = new ArrayList<>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Convert the arraylist to JSON
-    String json = convertToJson(messages);
+    String json = convertToJson(comments);
 
     // Send the JSON as the response
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String text = request.getParameter("text");
+    long timestamp = System.currentTimeMillis();
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("text", text);
+    commentEntity.setProperty("timestamp", timestamp);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+    // Redirect back to the HTML page.
```

Comment:
nit: ... and maybe this, as well. But it might be useful to have a comment explaining why you're redirecting (would it be bad if you didn't?) or where you're redirecting to.